### PR TITLE
Fixed #23242 -- Added ModelAdmin.approximate_date_hierarchy.

### PR DESCRIPTION
**Pull request by AlexProfi**
Updated from https://github.com/django/django/pull/5213
https://code.djangoproject.com/ticket/23242

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@ tests/coverage_html/
 tests/.coverage
 build/
 tests/report/
+.idea
```
Comment (timgraham) [inline .gitignore]: Revert

File: django/contrib/admin/templatetags/admin_list.py [hunk 2]
```diff
@@ -340,19 +340,33 @@ def date_hierarchy(cl):
...
         if not (year_lookup or month_lookup or day_lookup):
             # select appropriate start level
             date_range = cl.queryset.aggregate(first=models.Min(field_name),
                                                last=models.Max(field_name))
-            if date_range['first'] and date_range['last']:
-                if date_range['first'].year == date_range['last'].year:
-                    year_lookup = date_range['first'].year
-                    if date_range['first'].month == date_range['last'].month:
-                        month_lookup = date_range['first'].month
+            # if date_range['first'] and date_range['last']:
```
Comment (timgraham) [inline django/contrib/admin/templatetags/admin_list.py]: ?

Comment (AlexProfi) [inline django/contrib/admin/templatetags/admin_list.py]: I'll add docs
I'll try to add tests. at version 1.11 they were not work .
at version 1.9 I don't try to run it

**Comment by timgraham** [comment]
Please send a new pull request when it's completed.

**Pull request closed without merging**